dotCMS/src/main/java/com/dotmarketing/common/db/DotTimezonedTimestamp.java

```diff
@@ -0,0 +1,33 @@
+package com.dotmarketing.common.db;
+
+
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.base.Preconditions;
+
+import java.sql.Timestamp;
+import java.util.Date;
+
+public class DotTimezonedTimestamp {
+    private final String timezone;
+    private final Timestamp timestamp;
+
+    private DotTimezonedTimestamp(Date date, String timezone) {
+        Preconditions.checkArgument(UtilMethods.isSet(date), "Provided date cannot be null or empty");
+        Preconditions.checkArgument(UtilMethods.isSet(timezone), "Provided timezone cannot be null or empty");
+        this.timezone = timezone;
+        this.timestamp = new Timestamp(date.getTime());
+    }
+
+    public Timestamp getTimestamp() {
+        return timestamp;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Return a copy of "timestamp". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2384)